Packet++/src/DnsResource.cpp

```diff
@@ -56,7 +56,7 @@ size_t IDnsResource::decodeName(const char* encodedName, char* result, int itera
 		if ((wordLength & 0xc0) == 0xc0)
 		{
 			if (curOffsetInLayer + 2 > m_DnsLayer->m_DataLen || encodedNameLength > 255)
-				return encodedNameLength;
+				goto exit;
```

Comment:
`goto` is [not encouraged in c++](https://stackoverflow.com/questions/46586/goto-still-considered-harmful), I think we should try to avoid it...

Reply:
Usually it is true and I am also very against goto. But in this case it would mean copy/paste of the following code here and goto cleanup is the only acceptable pattern imho:
```cpp
        if (resultPtr > result)
	{
		result[resultPtr - result - 1] = 0;
	}
	if (resultPtr - result < 256)
	{
		// add the last '\0' to encodedNameLength
		resultPtr[0] = 0;
		encodedNameLength++;
	}
	return encodedNameLength;
```
Which makes more code duplication and less readable code.
Let me know what you think.

Reply:
We can move these lines to a new private method and then call it twice 🤔

Reply:
Ok, will do.